src/main/java/de/unijena/cheminf/clustering/desccalc/Descriptor.java

```diff
@@ -163,304 +1253,624 @@ public static int getNumberOfComponents(
         try {
             int tmpTotalNumberOfComponents = 0;
             for (Descriptor tmpDescriptor : aDescriptors) {
-                tmpTotalNumberOfComponents += descriptorToComponentNumberMap.get(tmpDescriptor);
+                tmpTotalNumberOfComponents += tmpDescriptor.getDescriptorComponentNumber();
             }
             return tmpTotalNumberOfComponents;
         } catch (Exception anException) {
             Descriptor.LOGGER.log(
                 Level.SEVERE,
-                "Descriptor.getNumberOfComponents: Number of components could not be evaluated: This should never happen."
+                "Descriptor.getNumberOfComponents: Number of components could not be evaluated: This should never happen.", anException
             );
-            throw new Exception("Descriptor.getNumberOfComponents: An exception occurred: This should never happen.");
+            throw new Exception("Descriptor.getNumberOfComponents: An exception occurred: This should never happen.", anException);
```

Comment:
This was probably Achim und not you but the general rule is to either log an exception or (re-)throw it. That is because the calling code above will also handle the exception (i.e. log it in most cases), so you don't need to do it here.